src/NUnitFramework/nunitlite/TextRunner.cs

```diff
@@ -361,12 +363,52 @@ private int ExploreTests()
                 runSettings[FrameworkPackageSettings.DefaultTestNamePattern] = options.DefaultTestNamePattern;
 
             if (options.TestParameters != null)
-                runSettings[FrameworkPackageSettings.TestParameters] = options.TestParameters;
+            {
+                var dictionary = new Dictionary<string, string>();
+
+                // Exactly what the framework used to do.
+                // Now can be independently changed without breaking backwards compatibility with old frameworks.
+                foreach (var param in options.TestParameters.Split(';'))
+                {
+                    var eq = param.IndexOf("=");
+
+                    if (eq > 0 && eq < param.Length - 1)
+                    {
+                        var name = param.Substring(0, eq);
+                        var val = param.Substring(eq + 1);
+
+                        dictionary[name] = val;
+                    }
+                }
+
+                SetTestParameters(runSettings, dictionary);
+            }
 
             return runSettings;
         }
 
         /// <summary>
+        /// Sets test parameters, handling backwards compatibility.
+        /// </summary>
+        private static void SetTestParameters(IDictionary<string, object> runSettings, IDictionary<string, string> testParameters)
+        {
+            runSettings[FrameworkPackageSettings.TestParametersDictionary] = testParameters;
+
+            if (testParameters.Count != 0)
+            {
+                // Kept for backwards compatibility with old frameworks.
+                // Reserializes the way old frameworks understand, even if the parsing above is changed.
+                // This reserialization cannot be changed without breaking compatibility with old frameworks.
+
+                var oldFrameworkSerializedParameters = new StringBuilder();
+                foreach (var parameter in testParameters)
```

Comment:
The standard is deliberately vague. IOW, we could have written "only use var if the rhs contains the name of the type" but we didn't. The idea (my idea) is that developers are supposed to use judgement - a notion that is too often forgotten. Whether it's sufficiently clear is a team judgement, since all code is clear to the guy who wrote it (at least for a while). My judgement is that it makes sense in context. Somebody else may say different.